### PR TITLE
feat: create custom link field

### PR DESCRIPTION
**Pull request by squigglybob**
add the ui for creating and displaying link fields

### CODE REVIEW

**Comment by squigglybob** [comment]
@corsacca I'm not sure I've dealt with the hidden link options in the correct way.

In terms of the data being returned it is not giving back any data for hidden link options.

In terms of the post_settings, the hidden link option is still in the `default` array and then I am using it's `deleted` property to not show that link option in the drop down.

Let me know if I should be filtering it out of the post_settings entirely or not...

**Review by corsacca — COMMENTED** [review]
Next places to review:
- Activity message 
image

- Details tile
image

- Displaying values in the list table
- Bug: Creating Links field as a private field doesn't keep values private.

File: dt-core/global-functions.php [hunk 1]
```diff
@@ -797,7 +804,68 @@ class="text-input"
...
+                    <div class="link-list-<?php echo esc_html( $field_key ) ?>">
+
+                        <?php
+                        foreach ( $post[$field_key] as $link_item ) {
+                            $option_type = $link_item["type"];
+                            $option_value = $fields[$field_key]["default"][$option_type];
+                            $meta_id = $link_item["meta_id"];
+                            $meta_value = $link_item["value"];
+
+                            render_link_field( $field_key, $option_key, $option_value, $meta_value, $display_field_id, $meta_id, $required_tag, $disabled );
```
[the remark this answers is not in the document]
Comment (squigglybob) [reply dt-core/global-functions.php]: fixed in [this commit](https://github.com/DiscipleTools/disciple-tools-theme/pull/1687/commits/872974f26471a270905a63665f3c4eca485478db)

File: dt-assets/js/details.js [hunk 2]
```diff
@@ -285,6 +347,27 @@ jQuery(document).ready(function($) {
     })
   })
 
+  /**
+   * Links
+   */
+  $('.add-link-form').on('submit', addLink)
+  function addLink(e) {
```
[the remark this answers is not in the document]
Comment (squigglybob) [reply dt-assets/js/details.js:354]: fixed in [this commit](https://github.com/DiscipleTools/disciple-tools-theme/pull/1687/commits/7bde94aee64524e82fb89314994870d5e2da4e96)

**Comment by corsacca** [comment]
- let's also add the ability to set links when creating a record

**Comment by squigglybob** [comment]
* [ ] Activity message
  image
* [ ] Details tile
  image
* [ ] Displaying values in the list table
* [ ] Bug: Creating Links field as a private field doesn't keep values private.
* [ ] let's also add the ability to set links when creating a record

**Comment by squigglybob** [comment]
@corsacca Not sure I understand what displaying values in the list table is referring to :O(
do you mean in the field explorer?

**Comment by squigglybob** [comment]
@corsacca ready for review 😄

**Comment by corsacca** [comment]
Suggestions for maybe separate Issue/PR

- Add button could be a dropdown with the types. 

- If there are no types should there be a message saying to create one?

- Auto add if there is only one selection


- Combine inputs under one label:
image

**Comment by corsacca** [comment]
Nice work @squigglybob !

**Pull request merged**